openmdao/core/problem.py

```diff
@@ -902,7 +906,7 @@ def final_setup(self):
         else:
             mode = self._orig_mode
 
-        if self._setup_status < 2:
+        if self._metadata['_setup_status'] < self.POST_FINAL_SETUP:
```

Comment:
Should we just go ahead and check here if setup status is 0, and if so call setup as well as final_setup instead of raising an exception if someone set the order after setup_procs (or changed any number of things that would also require a full setup)?

Reply:
Leaving this as is per our conversation with Justin and keeping the raise error